File: utils/move_to.py
from time import sleep

import requests


async def move_to(TOKEN, character, x, y):

    url = f"https://api.artifactsmmo.com/my/{character["name"]}/action/move"
    headers = {
        "Accept": "application/json",
        "Content-Type": "application/json",
        "Authorization": f"Bearer {TOKEN}"
    }

    body = {"x": x, "y": y}

    try:
        res = requests.post(url, headers=headers, json=body)
        data = res.json()

        if "error" in data:
            raise Exception(data["error"]["message"])

        destination = data["data"]["destination"]
        cooldown = data["data"]["cooldown"]

        print(f"✅ Moved to ({destination['x']}, {destination['y']}) on {destination['name']}")
        print(f"⏳ Cooldown started: {cooldown['total_seconds']} seconds")
        sleep(cooldown["total_seconds"])

    except Exception as e:
        print(f"❌ {e}")
        return False

    return True

File: src/utils/moveTo.ts
import type {Character} from "../@types/character";
import type {Location} from "../@types/location";
import {moveAction} from "../api/actions/move.ts";
import {log} from "../lib/logger.ts";
import {sleep} from "./sleep.ts";

export async function moveTo(character: Character, location: Location): Promise<Character> {
    try{
        const result = await moveAction(character.name, location.x, location.y);

        log(`Moved to (${result.destination.x}, ${result.destination.y}) on ${result.destination.name}`, "success", character.name);
        log(`Cooldown started: ${result.cooldown.total_seconds} seconds`, "info", character.name);

        await sleep(result.cooldown.total_seconds*1000);

        return result.character;
    } catch(err) {
        if (err instanceof DOMException && err.name === "AbortError") throw err;
        log(err instanceof Error ? err.message : String(err), 'error', character.name);
        return character;
    }
}